homeassistant/components/philips_air_purifier/__init__.py

```diff
@@ -0,0 +1,37 @@
+"""The Philips Air Purifier integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .client import ReliableClient
+from .const import COAP_PORT, DOMAIN
+
+PLATFORMS: list[str] = ["fan"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Philips Air Purifier from a config entry."""
+
+    client = ReliableClient(hass.loop, entry.data["host"], port=COAP_PORT)
+    client.start()
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][entry.entry_id] = client
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    client = hass.data[DOMAIN][entry.entry_id]
+    client.stop()
+
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
```

Comment:
```suggestion
    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
        client = hass.data[DOMAIN][entry.entry_id]
        client.stop()
        hass.data[DOMAIN].pop(entry.entry_id)
```

Reply:
Nice, I wasn't aware of the `:=` operator 😃

Reply:
[`Status`](https://github.com/mfrister/phipsair/blob/0c148533d88f807a9b7640e7dffe4c04d944bbca/phipsair/purifier.py#L52-L78) contains all information about the state of the purifier, not only information about availability. The `_set_status` method is passed as a callback to the PersistentClient, so it can push status updates to the Fan. I've updated the docstring to make this more clear.

I already had `@property available()` implemented and it indeed seems to stop Home Assistant from calling the other methods, so I could remove the `if`s there (see below).